ignition: keep network around when umounting

tormath1 opened the pull request:
This keeps network in the umounting Ignition stage as Equinix Metal needs to POST a status update to Equinix Metal timeline at each stage.

This solves: https://github.com/flatcar/Flatcar/issues/1536

## Testing done

* Tested on Flatcar CI here: https://github.com/flatcar/scripts/pull/2308
* Upstream PR: https://github.com/coreos/ignition/pull/1940

chewi reviewed (approved):
I was worried this might involve something hacky. This is beautiful!

tormath1 committed:
dracut: add dependency network to ignition-mount.service

On some providers (like Equinix Metal), there is a network dependency
for the umount stage, network must be still around when ExecStop is
executed.

Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
